core/app/sorters/spree/products/sort.rb

```diff
@@ -0,0 +1,60 @@
+module Spree
+  module Products
+    class Sort
+      def initialize(scope, params, current_currency)
+        @scope    = scope
+        @sort     = params[:sort]
+        @currency = params[:currency] || current_currency
+      end
+
+      def call
+        products = updated_at(scope)
+        products = price(products)
+
+        products
+      end
+
+      private
+
+      attr_reader :sort, :scope, :currency
+
+      def desc_order
+        @desc_order ||= String(sort)[0] == '-'
+      end
+
+      def sort_field
+        @sort_field ||= desc_order ? sort[1..-1] : sort
+      end
+
+      def updated_at?
+        sort_field == 'updated_at'
+      end
+
+      def price?
+        sort_field == 'price'
+      end
+
+      def order_direction
+        desc_order ? :desc : :asc
+      end
+
+      def arel_order(field)
+        Arel.sql("#{field} #{order_direction}")
+      end
+
+      def updated_at(products)
+        return products unless updated_at?
+
+        products.order(updated_at: order_direction)
+      end
+
+      def price(products)
+        return products unless price?
+
+        products.joins(master: :prices).
+          where(spree_prices: { currency: currency }).
+          order(arel_order("#{Spree::Price.table_name}.amount"))
```

Comment:
Can we avoid using `arel_order` here? Wouldn't `order("#{Spree::Price.table_name}.amount")` work?